vscode/src/chat/chat-view/sourcegraphRemoteFile.ts

```diff
@@ -0,0 +1,96 @@
+import { graphqlClient, isError } from '@sourcegraph/cody-shared'
+import { LRUCache } from 'lru-cache'
+import * as vscode from 'vscode'
+
+export class SourcegraphRemoteFileProvider implements vscode.FileSystemProvider, vscode.Disposable {
+    private cache = new LRUCache<string, string>({ max: 128 })
+    private disposables: vscode.Disposable[] = []
+
+    constructor() {
+        this.disposables.push(
+            vscode.workspace.registerFileSystemProvider('codysourcegraph', this, { isReadonly: true })
+        )
+    }
+
+    async readFile(uri: vscode.Uri): Promise<Uint8Array<ArrayBufferLike>> {
+        const content =
+            this.cache.get(uri.toString()) ||
+            (await SourcegraphRemoteFileProvider.getFileContentsFromURL(uri))
+
+        this.cache.set(uri.toString(), content)
+
+        return new TextEncoder().encode(content)
```

Comment:
I don't know how `readFile` is called, in theory it's possible that it's called twice (with `await`) with the same URI which would cause two fetches of the file contents.

Instead of caching the content you can cache the promise to avoid that, and also cache the result of calling `TextEncode` (I assume; you'll have to update the type of the cache of course):

```suggestion
        const cachedResult = this.cache.get(uri.toString())
        if (cachedResult) {
            return cachedResult
        }
        
        const content = SourcegraphRemoteFileProvider.getFileContentsFromURL(uri).then(content => new TextEncoder().encode(content))
        this.cache.set(uri.toString(), content)

        return content
```